src/allmydata/storage/http_client.py

```diff
@@ -122,6 +116,17 @@ def request(
                 "X-Tahoe-Authorization",
                 b"%s %s" % (secret.value.encode("ascii"), b64encode(value).strip()),
             )
+
+        # Note we can accept CBOR:
+        headers.addRawHeader("Accept", CBOR_MIME_TYPE)
+
+        # If there's a request message, serialize it and set the Content-Type
+        # header:
+        if message_to_serialize is not None:
+            assert "data" not in kwargs
```

Comment:
Maybe this case is a ... TypeError?

Tahoe definitely sprinkles `assert` statements throughout its implementation but maybe that's a bad pattern (I always thought it was, anyway ... I know mypy loves asserts but I don't think mypy can make any sense of _this_ one?).